src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java

```diff
@@ -63,29 +91,99 @@ public CommandResult execute(Model model) throws CommandException {
         }
-        Task taskToAssignTo = lastShownTaskList.get(taskIndex.getZeroBased());
+        Task taskToModify = lastShownTaskList.get(taskIndex.getZeroBased());
 
         List<Person> lastShownPersonsList = model.getFilteredPersonList();
+
+        Set<Contact> contactsToAdd = getAllContacts(personAddIndexes, personAddNames, lastShownPersonsList);
+        Set<Contact> contactsToDelete = getAllContacts(personDeleteIndexes, personDeleteNames, lastShownPersonsList);
+
+        Set<Contact> contactSetToModify = new HashSet<>(taskToModify.getAssignedContacts());
+        for (Contact contactToAdd : contactsToAdd) {
+            if (contactSetToModify.contains(contactToAdd)) {
+                alreadyAddedContacts.add(contactToAdd);
+            }
+            contactSetToModify.add(contactToAdd);
+        }
+        for (Contact contactToAdd : contactsToDelete) {
+            if (!contactSetToModify.contains(contactToAdd)) {
+                notAddedContacts.add(contactToAdd);
+            }
+            contactSetToModify.remove(contactToAdd);
+        }
+
+        Task editedTask = new Task(taskToModify.getTitle(), taskToModify.getCompleted(), contactSetToModify);
+
+        model.setTask(taskToModify, editedTask);
+        model.updateFilteredTaskList(PREDICATE_SHOW_ALL_TASKS);
+
+        if (!invalidIndexes.isEmpty() || !invalidNames.isEmpty()
+                || !alreadyAddedContacts.isEmpty() || !notAddedContacts.isEmpty()) {
+            throw new CommandException(buildErrorMessage() + MESSAGE_PARTIAL_SUCCESS);
+        }
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS, taskIndex.getOneBased()));
+    }
+
+    private Set<Contact> personIndexesToContacts(Set<Index> personIndexes, List<Person> personList) {
         Set<Contact> assignedContacts = new HashSet<>();
         for (Index personIndex : personIndexes) {
-            if (personIndex.getZeroBased() >= lastShownPersonsList.size()) {
-                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            if (personIndex.getZeroBased() >= personList.size()) {
+                invalidIndexes.add(personIndex);
+                continue;
             }
             Contact contactToAssign =
-                new Contact(lastShownPersonsList.get(personIndex.getZeroBased()).getName().fullName);
+                    new Contact(personList.get(personIndex.getZeroBased()).getName().fullName);
             assignedContacts.add(contactToAssign);
         }
+        return assignedContacts;
+    }
 
-        Task editedTask = new Task(taskToAssignTo.getTitle(), taskToAssignTo.getCompleted(), assignedContacts);
+    private Set<Contact> personNamesToContacts(Set<String> personNames, List<Person> personList) {
+        Set<Contact> assignedContacts = new HashSet<>();
+        for (String personName : personNames) {
+            String matchingPersonsName = Contact.isInPersonsList(personList, personName);
```

Comment:
Maybe it would be better if the method returns a boolean, especially since the method name is as such. Then you could do 

```
if (Contact.isInPersonsList(personList, personName)) {
    assignedContacts.add(personName)
} else {
    invalidNames.add(personName)
}

Reply:
I chose to return a matchingPersonsName so that the name in the newly created contact will match the person name in terms of capitalization! Maybe I can keep this return type but rename the method so that it doesn't sound like a boolean?